src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs

```diff
@@ -1436,18 +1446,20 @@ public static FormulaValue Switch(EvalVisitor runner, SymbolContext symbolContex
         }
 
         // ForAll([1,2,3,4,5], Value * Value)
-        public static FormulaValue ForAll(EvalVisitor runner, SymbolContext symbolContext, IRContext irContext, FormulaValue[] args)
+        public static async ValueTask<FormulaValue> ForAll(EvalVisitor runner, SymbolContext symbolContext, IRContext irContext, FormulaValue[] args)
         {// Streaming 
             var arg0 = (TableValue)args[0];
             var arg1 = (LambdaFormulaValue)args[1];
 
-            var rows = LazyForAll(runner, symbolContext, arg0.Rows, arg1);
+            var rowsAsync = LazyForAll(runner, symbolContext, arg0.Rows, arg1);
 
             // TODO: verify semantics in the case of heterogeneous record lists
-            return new InMemoryTableValue(irContext, StandardTableNodeRecords(irContext, rows.ToArray()));
+            var rows = await Task.WhenAll(rowsAsync);
```

Comment:
I wonder if it messes with natural order of rows (likely not, but would be good to check)

Reply:
good point - will double check